src/instructlab/model/evaluate.py

```diff
@@ -213,29 +220,32 @@ def qa_pairs_to_qna_to_avg_scores(qa_pairs: list[dict]) -> dict:
     return qna_to_avg_scores
 
 
-def launch_server(
-    ctx,
-    model,
-    model_name,
-    max_workers,
-) -> tuple:
+def launch_server(ctx, model, model_name, max_workers, backend) -> tuple:
     # pylint: disable=import-outside-toplevel
     # First Party
-    from instructlab.model.backends import backends
-
-    if not ctx.obj.config.serve.backend:
-        ctx.obj.config.serve.backend = backends.VLLM
-    if ctx.obj.config.serve.backend == backends.VLLM:
-        ctx.obj.config.serve.vllm.vllm_args.extend(["--served-model-name", model_name])
-    elif ctx.obj.config.serve.backend == backends.LLAMA_CPP:
-        # mt_bench requires a larger context size
-        ctx.obj.config.serve.llama_cpp.max_ctx_size = 5120
+
+    # we shouldn't be in the business of editing a user's cfg.
+    # with eval, this is tricky since there are a few things we want to help them out with if they do not exist.
+    # if the user did not set a backend, or model_path, set that for them. Everything else needs to be in the cfg. or in the cmd as flag overrides.
+    temp_serve = copy(ctx.obj.config.serve)
+    if backend is None:
+        temp_serve.backend = backends.VLLM
+    if backend == backends.VLLM:
+        temp_serve.vllm.vllm_args.extend(["--served-model-name", model_name])
+    if backend == backends.LLAMA_CPP:
+        if ctx.obj.config.serve.llama_cpp.max_ctx_size < 5120:
```

Comment:
I don't think a warning is the right way to handle this.  5120 is a requirement of eval.  It shouldn't matter what the user has set in config.

Reply:
ok, I am going to put a pretty specific debug (not warn) msg that we are going to change what is in the cfg

Reply:
How about:

"Evaluate requires a context size of >= 5120, ignoring serve configuration for max_ctx_size"

since we aren't asking the user to change their values.  A lower number might make sense for their use case but not for eval.